Cargo.toml

```diff
@@ -14,29 +14,22 @@ edition = "2018"
 travis-ci = { repository = "vorner/signal-hook" }
 maintenance = { status = "actively-developed" }
 
-[features]
-mio-support = ["mio"]
-mio-0_7-support = ["mio_0_7"]
-tokio-support = ["futures", "mio-support", "tokio-reactor"]
-
 [workspace]
 members = [
     "./",
     "signal-hook-registry",
+    "signal-hook-tokio-0_1",
+    "signal-hook-mio",
+    "signal-hook-mio-0_6",
```

Comment:
My idea was to have a crate without a version which will always use the latest version of mio or tokio. Only if there is a breaking change in mio or tokio we would add a new crate with a version to provide backward compatibility for clients not willing to switch to the new version. There is no signal-hook-tokio at the moment because tokio is already at 0.2 which is a breaking. I would like to provide a signal-hook-tokio for tokio 0.2 in another future PR. Then it will be signal-hook-tokio, signal-hook-tokio-0_1, signal-hook-mio and signal-hook-mio-0_6.

So if mio 0.8 requires breaking changes in the adapter library we will change signal-hook-mio to use mio 0.8 and introduce a signal-hook-mio-0_7 crate for clients which are stuck with mio 0.7.

Reply:
:thinking: That's an interesting approach.

Any (semi-)major version bump is breaking simply because the adaptor crate can't express a range of versions as dependency ‒ I believe it is not possible to say something like `mio-0.8.*..mio-0.9.*`.

With your approach, I worry a bit about discoverability. If I decide I want to stay on tokio 0.1 and not update to tokio 0.2, I'd need to update my dependency from signal-hook-tokio to signal-hook-tokio-0_1 at a point in time where I don't want to change anything, just stay up to date with something else. But how do I know I want to do that?

On the other hand, if each adaptor crate starts with the version number in it already, I fix it to the one I'll need for ever. When I want to upgrade tokio and I get compilation error because they don't match, looking into the Cargo.toml I'll naturally try changing that number.

Other option would be to do whatever here in the repository, but publish all under the same name, just different versions. But I don't know if this is well supported, as a version might need to be bumped in incompatible way when:
* signal-hook releases breaking change
* the adapted-to crate (tokio, ...) releases a breaking change
* the crate itself releases a breaking change

I don't know if all this can reasonably be encoded into the version.